classes/shop/Shop.php

```diff
@@ -441,7 +441,7 @@ public static function initialize()
                 $redirect_code = ($redirect_type == 1 ? '302' : '301');
                 $redirect_header = ($redirect_type == 1 ? 'Found' : 'Moved Permanently');
                 header('HTTP/1.0 ' . $redirect_code . ' ' . $redirect_header);
-                header('Location: http://' . $url);
+                header('Location: ' . Tools::getShopProtocol() . $url);
```

Comment:
To trigger this code, call a shop with URL rewriting enabled through a URL generated without GET parameters.

From instance: Calling `http://<shop_domain>/index.php?idPayment=1337&fc=module&module=cheque&controller=validation&id_lang=1` should trigger this code to redirect you to `http://<shop_domain>/en/module/cheque/validation?idPayment=1337`